expression/builtin_cast_vec.go

```diff
@@ -583,11 +585,31 @@ func (b *builtinCastTimeAsDurationSig) vecEvalDuration(input *chunk.Chunk, resul
 }
 
 func (b *builtinCastDurationAsDurationSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	var err error
+	if err = b.args[0].VecEvalDuration(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	res := result.GoDurations()
+	dec := b.tp.Decimal
+	var zeroT time.Time
+	for i, v := range res {
+		if result.IsNull(i) {
+			continue
+		}
+		dec, err := types.CheckFsp(dec)
```

Comment:
Can this line be moved to line 599?

Reply:
ok , i have moved "types.CheckFsp(dec)" out of loop